src/uu/users/src/users.rs

```diff
@@ -5,19 +5,97 @@
 
 // spell-checker:ignore (paths) wtmp
 
+use std::ffi::OsString;
+use std::path::Path;
+
 use clap::builder::ValueParser;
 use clap::{crate_version, Arg, Command};
+use uucore::error::UResult;
 use uucore::{format_usage, help_about, help_usage};
 
-mod platform;
+#[cfg(target_os = "openbsd")]
+use utmp_classic::{parse_from_path, UtmpEntry};
+#[cfg(not(target_os = "openbsd"))]
+use uucore::utmpx::{self, Utmpx};
 
 const ABOUT: &str = help_about!("users.md");
 const USAGE: &str = help_usage!("users.md");
 
 static ARG_FILES: &str = "files";
 
+fn get_long_usage() -> String {
+    #[cfg(not(target_os = "openbsd"))]
+    let default_path: &str = utmpx::DEFAULT_FILE;
+    #[cfg(target_os = "openbsd")]
+    let default_path: &str = "/var/log/wtmp";
+    format!(
+        "Output who is currently logged in according to FILE.
+If FILE is not specified, use {}.  /var/log/wtmp as FILE is common.",
+        default_path
+    )
+}
+
 #[uucore::main]
-use platform::uumain;
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let matches = uu_app()
+        .after_help(get_long_usage())
+        .try_get_matches_from(args)?;
+
+    let files: Vec<&Path> = matches
+        .get_many::<OsString>(ARG_FILES)
+        .map(|v| v.map(AsRef::as_ref).collect())
+        .unwrap_or_default();
+
+    let mut users: Vec<String>;
+
+    // OpenBSD uses the Unix version 1 UTMP, all other Unixes use the newer UTMPX
+    #[cfg(target_os = "openbsd")]
+    {
+        let filename = if files.is_empty() {
+            Path::new("/var/run/utmp")
+        } else {
+            files[0]
+        };
+        let entries = match parse_from_path(&filename) {
+            Ok(data) => data,
+            Err(_) => Vec::new(),
+        };
```

Comment:
I think it is more idiomatic to use `unwrap_or` instead of a `match`.
```suggestion
        let entries = parse_from_path(&filename).unwrap_or(Vec::new());
```

Reply:
Done as suggested.